plugins/SlicerT/SlicerTView.cpp

```diff
@@ -43,57 +49,63 @@ namespace lmms {
 	, m_slicerTParent(instrument)
+	, m_fullLogo(PLUGIN_NAME::getIconPixmap("full_logo"))
+	, m_background(PLUGIN_NAME::getIconPixmap("toolbox"))
 {
 	// window settings
 	setAcceptDrops(true);
 	setAutoFillBackground(true);
 
-	// render background
-	QPalette pal;
-	pal.setBrush(backgroundRole(), PLUGIN_NAME::getIconPixmap("artwork"));
-	setPalette(pal);
+	setMaximumSize(QSize(10000, 10000));
+	setMinimumSize(QSize(516, 400));
+	setResizable(true);
 
 	m_wf = new SlicerTWaveform(248, 128, instrument, this);
-	m_wf->move(2, 6);
+	m_wf->move(0, s_topBarHeight);
 
 	m_snapSetting = new ComboBox(this, tr("Slice snap"));
 	m_snapSetting->setGeometry(185, 200, 55, ComboBox::DEFAULT_HEIGHT);
 	m_snapSetting->setToolTip(tr("Set slice snapping for detection"));
 	m_snapSetting->setModel(&m_slicerTParent->m_sliceSnap);
 
-	m_syncToggle = new LedCheckBox("Sync", this, tr("SyncToggle"), LedCheckBox::LedColor::Green);
-	m_syncToggle->move(135, 187);
+	m_syncToggle = new PixmapButton(this, tr("Sync sample"));
+	m_syncToggle->setActiveGraphic(PLUGIN_NAME::getIconPixmap("sync_active"));
+	m_syncToggle->setInactiveGraphic(PLUGIN_NAME::getIconPixmap("sync_inactive"));
+	m_syncToggle->setCheckable(true);
 	m_syncToggle->setToolTip(tr("Enable BPM sync"));
 	m_syncToggle->setModel(&m_slicerTParent->m_enableSync);
 
 	m_bpmBox = new LcdSpinBox(3, "19purple", this);
-	m_bpmBox->move(130, 201);
 	m_bpmBox->setToolTip(tr("Original sample BPM"));
 	m_bpmBox->setModel(&m_slicerTParent->m_originalBPM);
 
 	m_noteThresholdKnob = createStyledKnob();
-	m_noteThresholdKnob->move(10, 197);
 	m_noteThresholdKnob->setToolTip(tr("Threshold used for slicing"));
 	m_noteThresholdKnob->setModel(&m_slicerTParent->m_noteThreshold);
 
 	m_fadeOutKnob = createStyledKnob();
-	m_fadeOutKnob->move(64, 197);
 	m_fadeOutKnob->setToolTip(tr("Fade Out per note in milliseconds"));
 	m_fadeOutKnob->setModel(&m_slicerTParent->m_fadeOutFrames);
 
 	m_midiExportButton = new QPushButton(this);
-	m_midiExportButton->move(199, 150);
 	m_midiExportButton->setIcon(PLUGIN_NAME::getIconPixmap("copy_midi"));
 	m_midiExportButton->setToolTip(tr("Copy midi pattern to clipboard"));
 	connect(m_midiExportButton, &PixmapButton::clicked, this, &SlicerTView::exportMidi);
 
+	m_folderButton = new PixmapButton(this);
+	m_folderButton->setActiveGraphic(PLUGIN_NAME::getIconPixmap("folder_icon"));
+	m_folderButton->setInactiveGraphic(PLUGIN_NAME::getIconPixmap("folder_icon"));
+	m_folderButton->setToolTip(tr("Open sample selector"));
+	connect(m_folderButton, &PixmapButton::clicked, this, &SlicerTView::openFiles);
+
 	m_resetButton = new QPushButton(this);
-	m_resetButton->move(18, 150);
 	m_resetButton->setIcon(PLUGIN_NAME::getIconPixmap("reset_slices"));
-	m_resetButton->setToolTip(tr("Reset Slices"));
+	m_resetButton->setToolTip(tr("Reset slices"));
 	connect(m_resetButton, &PixmapButton::clicked, m_slicerTParent, &SlicerT::updateSlices);
+
+	update();
 }
 
 Knob* SlicerTView::createStyledKnob()
```

Comment:
It would be easier to add another knob style in Knob.cpp and use that here.

Reply:
How would that be any easier? Even if it were, the knob styles are supposed to be for general use, not something specific like knobs used by only one plugin.